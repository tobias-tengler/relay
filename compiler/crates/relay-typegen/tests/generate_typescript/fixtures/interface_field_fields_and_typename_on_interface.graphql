fragment b on Query {
  node(id: "") {
    __typename
    id
    ... on User {
      name
    }
    ... on Page {
      doesViewerLike
    }
  }
}