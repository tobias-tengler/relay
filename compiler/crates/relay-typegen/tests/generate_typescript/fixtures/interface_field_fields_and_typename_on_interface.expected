==================================== INPUT ====================================
fragment b on Query {
  node(id: "") {
    __typename
    id
    ... on User {
      name
    }
    ... on Page {
      doesViewerLike
    }
  }
}
==================================== OUTPUT ===================================
import { FragmentRefs } from "relay-runtime";
export type b$data = {
  readonly node: {
    readonly __typename: string;
    readonly doesViewerLike?: boolean | null | undefined;
    readonly id: string;
    readonly name?: string | null | undefined;
  } | null | undefined;
  readonly " $fragmentType": "b";
};
export type b$key = {
  readonly " $data"?: b$data;
  readonly " $fragmentSpreads": FragmentRefs<"b">;
};
